### PR TITLE
Update SDL2 used from source to v2.24.1

### PR DESCRIPTION
**Pull request by tsunamistate**
VCPKG already updated
Release notes: https://github.com/libsdl-org/SDL/releases/tag/release-2.24.1

### CODE REVIEW

File: 3rdParty/SDL2/CMakeLists.txt [hunk 1]
```diff
@@ -15,7 +15,7 @@ set(SDL_TEST_ENABLED_BY_DEFAULT OFF)
 include(functions/FetchContent_MakeAvailableExcludeFromAll)
 include(FetchContent)
 FetchContent_Declare(SDL2
-    URL https://github.com/libsdl-org/SDL/archive/cb46e1b3f06a08d57b4ccd83127d1ec3139e1c0f.tar.gz
-    URL_HASH MD5=b5539b578ef77f6364f621dc55c244d7
+    URL https://github.com/libsdl-org/SDL/archive/a1d1946dcba6509f0679f507b57e7b228d32e6f8.tar.gz
```
Comment (glebm) [inline 3rdParty/SDL2/CMakeLists.txt:18]: Do we also need to update the Android project files?

Comment (tsunamistate) [reply 3rdParty/SDL2/CMakeLists.txt:18]: I don't think so, android-project didn't change from 2.24.0 to 2.24.1: https://github.com/libsdl-org/SDL/commits/release-2.24.x/android-project

Comment (tsunamistate) [reply 3rdParty/SDL2/CMakeLists.txt:18]: Also, no changes on Android in 2.24.1 release notes

Comment (AJenbo) [reply 3rdParty/SDL2/CMakeLists.txt:18]: There are a few minor fixes and gamepad support additions that where skipped when i upgraded from 2.0.22 to 2.0.23, not that it should hold back this PR

**Review by AJenbo — APPROVED** [review]

**Pull request merged**